examples/modular-drivers/src/free_heap_sensor.rs

```diff
@@ -21,7 +21,7 @@ use micro_rdk::{
 pub struct FreeHeapSensor;
 
 pub fn register_models(registry: &mut ComponentRegistry) -> Result<(), RegistryError> {
-    registry.register_sensor("free-heap", &FreeHeapSensor::from_config)?;
+    registry.register_sensor("free-heap".to_string(), &FreeHeapSensor::from_config)?;
```

Comment:
Any existing modules will have their registration calls broken by this change. See below in `registry` for a suggestion on how to avoid that.

Reply:
reverted examples, though now compatible with both string slices and Strings